src/jimgw/core/single_event/detector.py

```
@@ -500,27 +496,30 @@ def load_and_set_psd(self, psd_file: str = "", asd_file: str = "") -> PowerSpect
-            f, psd_vals = loadtxt(psd_file, unpack=True)
-        elif asd_file != "":
-            f, asd_vals = loadtxt(asd_file, unpack=True)
-            psd_vals = asd_vals**2
+        if psd_file:
+            _loaded_psd = PowerSpectrum.from_file(psd_file, is_asd=False)
+        elif asd_file:
+            _loaded_psd = PowerSpectrum.from_file(asd_file, is_asd=True)
         else:
             logger.info("Grabbing GWTC-2 PSD for " + self.name)
             url = asd_file_dict[self.name]
             data = requests.get(url)
```

Ruff (0.15.15)

[error] Probable use of `requests` call without timeout

(S113)